src/wayland/buffer/mod.rs

```diff
@@ -16,8 +16,15 @@ use wayland_server::protocol::wl_buffer;
 /// [`DelegateDispatch`](wayland_server::DelegateDispatch) implementations to notify the compositor when a
 /// buffer is destroyed.
 pub trait BufferHandler {
+    /// Called when a buffer is attached to a surface.
```

Comment:
I think we could make this comment more detailed, maybe adding something like

> Note that at this point the buffer has not yet been committed to the surface, and it is possible that it will never be. This callback is meant to handle preparatory work to handle the buffer (such as tracking DMA fences), but should not be used as an indication that the contents of that buffer will be displayed for sure, nor about when they will be displayed.